framework/src/distributions/Distribution.C

```diff
@@ -13,28 +13,45 @@
 /****************************************************************/
 
 #include "Distribution.h"
+#include "MooseRandom.h"
 
 template<>
 InputParameters validParams<Distribution>()
 {
   InputParameters params = validParams<MooseObject>();
-  params.addParam<unsigned int>("seed", _default_seed, "Random number generator seed");
-  params.addRequiredParam<std::string>("type", "Distribution type");
+  params.addParam<unsigned int>("seed", 10318691, "Random number generator seed");
   params.registerBase("Distribution");
-  params.declareControllable("enable");
+
   return params;
 }
 
 Distribution::Distribution(const InputParameters & parameters)
- : MooseObject(parameters),
-   SetupInterface(this)
+  : MooseObject(parameters),
+    _seed(getParam<unsigned int>("seed")),
+    _random(new MooseRandom())
```

Comment:
Why is this dynamically allocated? Doing `MooseRandom _random` in the header file does not work?